M2/Macaulay2/packages/FastLinAlg.m2

```diff
@@ -634,43 +676,45 @@ nonzeroEntries (Matrix):= opts -> (M1) ->(
 
 
 
---this function checks regularInCodimension via reduction mod p
-regularInCodimensionReductionP = method(Options=>optRegularInCodimension);
-regularInCodimensionReductionP(ZZ, Ring, ZZ):= opts -> (n1, R1, p)-> (
+--this function checks Rn via reduction mod p
+RnReductionP = method(Options=>optRn);
+RnReductionP(ZZ, Ring, ZZ):= opts -> (n1, R1, p)-> (
     ambR := ambient R1;
     genList := generators(ambR);
     ambRing := ZZ/p[genList];
 
     Id := sub(ideal R1, ambRing);
 --    myHash := new HashTable from opts;
 
-    return regularInCodimension(n1, ambRing/Id, opts);
+    return Rn(n1, ambRing/Id, opts);
 );
 
 verifyStrategy := (passedStrat) -> (
-    (
-        instance(passedStrat, HashTable) 
-        and (passedStrat #? LexLargest) 
-        and (passedStrat #? LexSmallestTerm)
-        and (passedStrat #? LexSmallest)
-        and (passedStrat #? GRevLexSmallestTerm)
-        and (passedStrat #? GRevLexSmallest)
-        and (passedStrat #? GRevLexLargest)
-        and (passedStrat #? Random)
-        and (passedStrat #? RandomNonzero)
-    )
+    if not ((class passedStrat === HashTable) or ((class passedStrat) === MutableHashTable)) then return false;
```

Comment:
This doesn't look like an improvement.  What's wrong with the previous code?  It's compact and clear.  It could even be clearer if 16 parentheses were removed.

Reply:
I think I've fixed it.  It will be pushed shortly.